src/vs/workbench/services/preferences/browser/preferencesService.ts

```diff
@@ -240,7 +240,7 @@ export class PreferencesService extends Disposable implements IPreferencesServic
 	}
 
 	private async openSettings2(options: IOpenSettingsOptions): Promise<IEditorPane> {
-		const input = this.settingsEditor2Input;
+		const input = this._register(this.settingsEditor2Input);
```

Comment:
Is there a concern with multiple instances of settingsEditor2Input being retained for each call to `openSettings2`? As-is ever input will be retained for the lifetime of the PreferencesService

Reply:
I think the disposable should not be registered here, since it's passed to editorService.openEditor, which should then own the editor's lifetime (I assume).

Maybe the TestEditorService.openEditor should just dispose the editor it's been given?

Reply:
I tried changing around a few things but it looks like
1. I can't dispose the input immediately after using it in the openEditor call because it sometimes results in the Settings editor closing right after being opened. I can't use a mutable disposable for it either for a similar reason while switching between the JSON and UI editors.
2. Even after wrapping TestEditorService#openEditor to hold onto and then dispose the inputs, there's still a PreferencesService leak. I'm unsure whether it's caused by the test framework or not. There _is_ an instantiation in PreferencesService that requires a new IPreferencesService to be instantiated, instead of `this` being passed in, and I wouldn't be surprised if that's causing a circular reference: https://github.com/microsoft/vscode/blob/c47d950e7b974083e52a3e0853167dd2f5dbcd81/src/vs/workbench/services/preferences/test/browser/preferencesService.test.ts#L37 and https://github.com/microsoft/vscode/blob/c47d950e7b974083e52a3e0853167dd2f5dbcd81/src/vs/workbench/services/preferences/browser/preferencesService.ts#L109

Reply:
Point 2 is solved. I verified that it's a singleton, so I'll just pass in `this`.
Point 1 is just fyi.